storage/redis/peer_store.go

```diff
@@ -619,42 +619,46 @@ func (ps *peerStore) AnnouncePeers(ih bittorrent.InfoHash, seeder bool, numWant
 	return
 }
 
-func (ps *peerStore) ScrapeSwarm(ih bittorrent.InfoHash, af bittorrent.AddressFamily) (resp bittorrent.Scrape) {
+func (ps *peerStore) ScrapeSwarms(infoHashes []bittorrent.InfoHash, af bittorrent.AddressFamily) (resp []bittorrent.Scrape) {
 	select {
 	case <-ps.closed:
 		panic("attempted to interact with stopped redis store")
 	default:
 	}
 
-	resp.InfoHash = ih
-	addressFamily := af.String()
-	encodedInfoHash := ih.String()
-	encodedLeecherInfoHash := ps.leecherInfohashKey(addressFamily, encodedInfoHash)
-	encodedSeederInfoHash := ps.seederInfohashKey(addressFamily, encodedInfoHash)
+	resp = make([]bittorrent.Scrape, len(infoHashes))
 
 	conn := ps.rb.open()
 	defer conn.Close()
 
-	leechersLen, err := redis.Int64(conn.Do("HLEN", encodedLeecherInfoHash))
-	if err != nil {
-		log.Error("storage: Redis HLEN failure", log.Fields{
-			"Hkey":  encodedLeecherInfoHash,
-			"error": err,
-		})
-		return
-	}
+	for i, ih := range infoHashes {
```

Comment:
So redis doesn't support fullscrapes? :(

Reply:
It's been a while since I wrote this, but my gut feeling is that this had some complexity due to the possibility of asynchronous stuff happening.. I'd probably have to design another `WATCH ... MULTI ... EXEC` block and prove that to be correct. What's the priority on this? How about just a note in the docs that redis currently doesn't do fullscrapes, and an issue to keep track of it?